spark/v3.3/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSqlExtensionsAstBuilder.scala

```diff
@@ -267,6 +286,12 @@ class IcebergSqlExtensionsAstBuilder(delegate: ParserInterface) extends IcebergS
   private def typedVisit[T](ctx: ParseTree): T = {
     ctx.accept(this).asInstanceOf[T]
   }
+
+  private def timeRetain(unit: String, duration: Long): Long = unit.toUpperCase(Locale.ENGLISH) match {
+    case "MONTHS" => 30 * TimeUnit.DAYS.toMillis(duration)
```

Comment:
hmm, good point, there is not exactly always 30 days in a month. That's probably why it is not supported by `TimeUnit`. So I would say in that case let's remove MONTHS from the time unit in parser and just support DAYS, HOURS, MINUTES and SECONDS

Reply:
Yes, it does. Updated.